src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T item, Node<T> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        Node<T> last = tail;
+        Node<T> newNode = new Node<>(last, value, null);
+        tail = newNode;
+        if (last == null) {
+            head = newNode;
+        } else {
+            last.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (size == index) {
+            add(value);
+        } else {
+            checkIndex(index);
+            Node<T> indexNode = searchNode(index);
+            Node<T> newNode = new Node<>(indexNode.prev, value, indexNode);
+            if (indexNode.prev == null) {
+                head = newNode;
+            } else {
+                indexNode.prev.next = newNode;
+            }
+            indexNode.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list != null) {
+            for (T fromList : list) {
+                Node<T> newNode = new Node<>(tail, fromList, null);
+                tail.next = newNode;
+                tail = newNode;
+                size++;
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return searchNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> foundNode = searchNode(index);
+        T deletedValue = foundNode.item;
+        foundNode.item = value;
+        return deletedValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> foundNode = searchNode(index);
+        return deleteLink(foundNode);
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> i = head; i != null; i = i.next) {
+                if (i.item == null) {
+                    deleteLink(i);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> i = head; i != null; i = i.next) {
```

Comment:
make it an `if` without `else` with a `object == i.item || object != null && object.equals(i.item)` condition